framework/fp_vectors/Floats.h

```diff
@@ -98,28 +135,31 @@ extern "C" {
 struct Float16
 {
     union {
-#if defined(SANDSTONE_HAS_FLOAT16_TYPE) || \
-    defined(SANDSTONE_HAS__FP16_TYPE)
-
-        fp16_t as_float;
-
-#endif
-
-        union {
-            uint16_t as_hex;
-            uint16_t payload;
+        struct __attribute__((packed)) {
+            uint16_t mantissa : FP16_MANTISSA_BITS;
+            uint16_t exponent : FP16_EXPONENT_BITS;
+            uint16_t sign     : FP16_SIGN_BITS;
         };
 
         struct __attribute__((packed)) {
-            uint16_t mantissa : 10;
-            uint16_t exponent : 5;
-            uint16_t sign : 1;
-        };
+            uint16_t payload  : FP16_MANTISSA_BITS - 1;
+            uint16_t quiet    : 1;
+            uint16_t exponent : FP16_EXPONENT_BITS; // BFLOAT16_NAN_EXPONENT
+            uint16_t sign     : FP16_SIGN_BITS;
```

Comment:
`quiet` field uses `1` and `sign` uses special constant `FP16_SIGN_BITS`.
This is inconsistent. Let's do everything one way or the other.